src/web/components/Search/SearchBar.tsx

```diff
@@ -6,24 +6,26 @@ import './SearchBar.scss';
 
 type SearchBarContainerProps = React.PropsWithChildren<{
   className?: string;
-  handleOnBlur: () => void;
+  handleOnBlur?: () => void;
 }>;
 export function SearchBarContainer({ children, className, handleOnBlur }: SearchBarContainerProps) {
   const componentRef = useRef<HTMLDivElement>(null);
-  const handleClick = (event: MouseEvent) => {
-    if (componentRef.current && !componentRef.current.contains(event.target as Node)) {
-      handleOnBlur();
-    }
-  };
 
   useEffect(() => {
+    const handleClick = (event: MouseEvent) => {
+      if (componentRef.current && !componentRef.current.contains(event.target as Node)) {
+        if (handleOnBlur) {
```

Comment:
could we check this before we add the event listener?

Reply:
great catch - does that look right now?